src/clib/xtg/surf_xyz_from_ij.c

```diff
@@ -119,7 +119,8 @@ surf_xyz_from_ij(int i,
     /* if (xdist<0 && ydist<0)  beta=2*PI - beta; */
     /* if (xdist>=0 && ydist<0) beta=PI + beta; */
 
-    if (beta < 0 || beta > PI / 2.0) {
+    if (beta < 0 || beta > PI / 2.0 || isnan(beta)) {
```

Comment:
This check should probably be moved to after x and y have definitely been initialized as they are outputs of the function on the python side.